### DISCUSSION TITLE
Actividad 4 Grupo 13

### DISCUSSION THREAD

**Discussion by pilita-acosta** [frlp-utn-ingsoft/catedra#169] [Show and tell]
https://github.com/facularrauri/vetsoft/tree/v1.0.3

**Comment by ivanduva** [comment]
Buenas.
La entrega está fuera de término. Recuerden que como explicamos en las condiciones de aprobación, si entregan fuera de término 2 veces quedan afuera de la promoción. Por ahora tienen 1.

Podrían pasarnos el link a donde deployaron?

Saludos

**Reply by pilita-acosta** [reply]
Hola Ivan, estuvimos 2 integrantes del equipo enfermos por más de una semana y no pudimos terminarlo en tiempo y forma, le enviamos los certificados a Rodrigo por email pidiendo una prórroga en la entrega y nos respondió que sí.
En cuanto al link te lo envío por acá, también lo podes encontrar en nuestro repositorio en la sección "About".
Saludos!


https://vetsoft-5xwo.onrender.com/

**Reply by ivanduva** [reply]
Listo, perfecto entonces.
Saludos

**Reply by ivanduva** [reply]
Buenas,
En general hay un buen trabajo y varios puntos altos en el trabajo, pero les vamos a pedir algunos arreglos sobre todo porque vemos algunos problemas de procedimiento:

- En cuanto a la validación de que el teléfono sea numérico, no vemos ningún test hecho. Les pedimos que agreguen uno de unidad, uno de integración y otro e2e como hicieron en las demás tareas.
- En cuanto a la validación de que el teléfono empiece con 54, el test e2e chequea que aparezca el error en caso de que el teléfono no sea numérico. Debería testear que tire el error de que no empieza con 54.
- En cuanto a la validación del email, se permite agregar "@vetsoft.com" sin nada adelante.
- Luego de arreglar todo, saquen una release.

Recuerden que los cambios deben ser realizados por las personas que se encargaron de cada issue.

Después les dejamos algunos comentarios sobre procedimiento que deberían revisar en las próximas correcciones:

- En las reviews no hay comentarios con relevancia. Como dijimos más de una vez, comentarios del estilo "ok" o "Funciona bien" no aportan a la review. Para eso se pone el approve y listo.
- Vimos que hicieron commits sobre main para agregar lo relativo a la CI/CD. Si estamos usando el workflow Feature Branch, esto es peligroso y constituye una muy mala práctica. Estos commits no pasan por una instancia formal de review y por lo tanto es poco seguro, puede romper la aplicación. Hay que hacer PRs para cumplir con el proceso.
- Vemos varios malos mensajes de commit, en algunos casos son poco explicativos y en otros se mencionan cuestiones relativas al código o a los archivos, lo cual es redundante porque esa info ya está en el commit. Recuerden que un mensaje de commit debería contener una síntesis de lo que ese commit soluciona en cuanto a funcionalidad.

Por otro lado, hay un muy buen trabajo en los tests. Están hechos de manera muy criteriosa. También las releases están muy bien documentadas. También valoramos que se hayan arreglado para hacer el deploy en vercel y que hayan organizado las github actions.  En ese sentido sigan así.

Arreglando lo anterior ya tendrían el trabajo integrador completamente aprobado.

**La última fecha límite de será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by pilita-acosta** [reply]
Buenas tardes profe, ya publicamos la ultima release 
https://vetsoft-5xwo.onrender.com/

**Comment by ivanduva** [comment]
Buenas,
El trabajo integrador queda aprobado.
Igualmente no queriamos dejar de hacerles un comentario para que les sirva en el futuro: vimos que en algunos PR (por ejemplo este https://github.com/facularrauri/vetsoft/pull/19) donde una persona que no participaba del PR resuelve los conflictos antes de mergear. Primero, esto no es un comportamiento común. En general la persona que viene trabajando sobre ese PR es quien resuelve los conflictos. Y segundo, la instancia de resolución de conflictos es un proceso clave en el uso de git y que requiere de mucha práctica para acostumbrarse a hacerlo bien. Por eso les recomendamos que no le esquiven a esto porque es algo que muy seguramente van a tener que hacer en un trabajo en la industria y hacerlo mal les puede traer problemas.

Saludos
